imports/plugins/included/jobcontrol/server/jobs/cart.js

```diff
@@ -0,0 +1,91 @@
+import later from "later";
+import moment from "moment";
+import { Accounts, Cart, Jobs } from "/lib/collections";
+import { Hooks, Logger, Reaction } from "/server/api";
+
+const olderThan = moment().subtract(3, "days")._d;
+
+Hooks.Events.add("onJobServerStart", () => {
+  Logger.debug("Adding Job removeStaleCart to jobControl");
+  const staleCarts = Cart.find({
+    updatedAt: {
+      $lte: olderThan
+    }
+  }).fetch();
+  if (staleCarts) {
+    new Job(Jobs, "cart/removeFromCart", {})
+      .priority("normal")
+      .retry({
+        retries: 5,
+        wait: 60000,
+        backoff: "exponential" // delay by twice as long for each subsequent retry
+      })
+      .repeat({
+        schedule: later.parse.text("every day")
+      })
+      .save({
+        cancelRepeats: true
+      });
+  } else {
+    Logger.info("No stale carts at the moment");
+  }
+});
+
+export default () => {
+  const removeStaleCart = Jobs.processJobs("cart/removeFromCart", {
+    pollInterval: 60 * 60 * 1000, // backup polling, see observer below
+    workTimeout: 180 * 1000
+  }, (job, callback) => {
+    Logger.debug("Processing cart/removeFromCart");
+    const cartDetails = Cart.find({
+      updatedAt: {
+        $lte: olderThan
+      }
+    }).fetch();
+    cartDetails.map(cart => {
+      if (cart.items) {
+        const user = Accounts.findOne({ userId: cart.userId });
+        if (!user.emails.length) {
+          const result = Cart.update(
+            { userId: user._id },
```

Comment:
By "cleaning the cart" I think we meant removing the cart record. We should also be removing stale Account and Session records as noted in the ticket.